odoo/tools/image.py

```diff
@@ -32,89 +33,122 @@
 IMAGE_MAX_RESOLUTION = 45e6
 
 
-# ----------------------------------------
-# Image resizing
-# ----------------------------------------
-
-def image_resize_image(base64_source, size=IMAGE_BIG_SIZE, filetype=None):
-    """ Function to resize an image. The image will be resized to the given
-        size, while keeping the aspect ratios. The image will not be resized if
-        smaller than the expected size.
-
-        :param base64_source: base64-encoded version of the source
-            image; if False, returns False
-        :param size: 2-tuple(width, height). A None value for any of width or
-            height mean an automatically computed value based respectively
-            on height or width of the source image.
-        :param filetype: the output filetype, by default the source image's
-        :type filetype: str, any PIL image format (supported for creation)
-    """
-    if not base64_source:
-        return False
-    # Return unmodified content if no resize or we etect first 6 bits of '<'
-    # (0x3C) for SVG documents - This will bypass XML files as well, but it's
-    # harmless for these purposes
-    if base64_source[:1] == b'P':
-        return base64_source
-
-    image = base64_to_image(base64_source)
-    # store filetype here, as Image.new below will lose image.format
-    filetype = (filetype or image.format).upper()
-
-    filetype = {
-        'BMP': 'PNG',
-    }.get(filetype, filetype)
-
-    if size and (size[0] or size[1]):
-        w, h = image.size
-        asked_width = size[0] or (w * size[1]) // h
-        asked_height = size[1] or (h * size[0]) // w
-
-        image.thumbnail((asked_width, asked_height), Image.LANCZOS)
-
-    if image.mode not in ["1", "L", "P", "RGB", "RGBA"] or (filetype == 'JPEG' and image.mode == 'RGBA'):
-        image = image.convert("RGB")
-
-    return image_to_base64(image, filetype)
+def image_process(base64_source, size=(0, 0), verify_resolution=False, quality=80, crop=None, colorize=False, format=None):
```

Comment:
@JKE-be 

hard testing to do

Reply:
@JKE-be

hard testing to do

(so you get an email)